src/sql/workbench/services/executionPlan/common/executionPlanService.ts

```diff
@@ -27,21 +27,59 @@ export class ExecutionPlanService implements IExecutionPlanService {
 		this._providerRegisterEvent = this._onProviderRegister.event;
 	}
 
-	/**
-	 * Runs the actions using the provider that supports the fileFormat provided.
-	 * @param fileFormat fileformat of the underlying execution plan file. It is used to get the provider that support it.
-	 * @param action executionPlanService action to be performed.
-	 */
-	private async _runAction<T>(fileFormat: string, action: (handler: azdata.executionPlan.ExecutionPlanProvider) => Thenable<T>): Promise<T> {
+	private async ensureCapabilitiesRegistered(): Promise<void> {
 		let providers = Object.keys(this._capabilitiesService.providers);
 		if (!providers) {
-			providers = await new Promise(resolve => {
+			await new Promise<void>(resolve => {
 				this._capabilitiesService.onCapabilitiesRegistered(e => {
-					resolve(Object.keys(this._capabilitiesService.providers));
+					resolve();
 				});
 			});
 		}
+	}
+
+	private async getExecutionPlanProvider(providerId: string): Promise<azdata.executionPlan.ExecutionPlanProvider> {
```

Comment:
This should be returning `Promise<azdata.executionPlan.ExecutionPlanProvider | undefined>`